src/ethereum/EtherscanButton/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import { Icon } from '../..';
+
+type Network = 'mainnet' | 'rinkeby';
+
+type Props = {
+  className?: string;
+  network?: Network;
+  type: 'address' | 'tx';
+  value: string;
+};
+
+const getNetwork = (network: Network) => {
+  const lowerCaseNetwork = network.toLowerCase();
+  return lowerCaseNetwork === 'mainnet' ? '' : `${lowerCaseNetwork}.`;
+};
+
+const EtherscanButton = ({
+  className,
+  type,
+  value,
+  network = 'mainnet',
+}: Props): React.ReactElement => {
+  const getEtherscanLink = () =>
```

Comment:
Why this couldn't be just
```
const etherscanLink = `https://${getNetwork(network)}etherscan.io/${type}/${value}`
```

I've seen multiple places where you define value as `const getValue = () => value` and then use the function in render method and always wanted to ask this

Reply:
right, it was used for the button, now that we are using a link, it's no needed.